bucket/foobar2000.json

```
@@ -18,34 +18,35 @@
     "persist": [
         "configuration",
         "index-data",
-        "library"
+        "library",
+        "user-components"
     ],
     "installer": {
         "script": [
             "Remove-Item \"$dir\\`$*\", \"$dir\\uninstall.exe\" -Force -Recurse",
-            "New-Item \"$dir\\portable_mode_enabled\" -Force | Out-Null",
-            "if(!(Test-Path \"$persist_dir\\playlists\")) {",
-            "    New-Item \"$persist_dir\\playlists\" -ItemType Directory -Force | Out-Null",
+            "New-Item \"$dir\\portable_mode_enabled\" | Out-Null",
+            "if(!(Test-Path \"$persist_dir\")) {",
             "    if (Test-Path \"$env:Appdata\\foobar2000\") {",
             "        Write-Host -F yellow \"Copying old '$env:Appdata\\foobar2000' to '$persist_dir'\"",
-            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Recurse -Force",
+            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Recurse",
             "    }",
             "}",
             "else {",
-            "    $shortVersion = $version.Split('.')[0, 1] -Join '.'",
-            "    Copy-Item \"$persist_dir\\playlists\" \"$dir\\playlist-v$shortVersion\" -Force -Recurse",
+            "    Move-Item \"$persist_dir\\playlists*\" \"$dir\"",
             "}",
             "if (Test-Path \"$persist_dir\\theme.fth\") {",
-            "    Copy-Item \"$persist_dir\\theme.fth\" \"$dir\" -Force",
+            "    Copy-Item \"$persist_dir\\theme.fth\" \"$dir\"",
             "}"
         ]
     },
-    "post_install": "if (is_directory \"$dir\\theme.fth\") { Remove-Item \"$dir\\theme.fth\" -Force }",
     "uninstaller": {
         "script": [
-            "Copy-Item \"$dir\\playlists-v*\\*\" \"$persist_dir\\playlists\" -Force  -Recurse",
+            "$shortVersion = $version.Split('.')[0, 1] -Join '.'",
+            "if (Test-Path \"$dir\\playlists-v$shortVersion\") {",
+            "    Copy-Item \"$dir\\playlists-v$shortVersion\" \"$persist_dir\" -Recurse",
+            "}",
```

```suggestion
            "}",
            "else { Copy-Item \"$dir\\playlists*\" \"$persist_dir\" -Recurse }",
```

If there is no new playlist, just persist the old one.